### PR TITLE
Move Fluentd position files from /var/log to /var/run.

### PR DESCRIPTION
**Pull request by qingling128**

### CODE REVIEW

**Comment by qingling128** [comment]
It turned out that the `in_system` plugin does not support customizing the `pos_file`. As a result, I changed it to `storage`. However, we still can not make `/var/log` read only because Fluentd itself needs to write logs there.

**Comment by qingling128** [comment]
Tested:
image

And logs flow just fine:
image

**Comment by qingling128** [comment]
For example, these commands will fail to create `/var/log/google-fluentd` folder:
https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/templates/package-scripts/td-agent/rpm/post#L9

We could potentially change it. It does not seem worth it IMO though.

File: logging-agent.yaml [hunk 1]
```diff
@@ -408,43 +408,58 @@ data:
...
     # Whether to include node-journal or not is determined when starting the
     # cluster. It is not changed when the cluster is already running.
     <source>
       @type systemd
-      pos_file /var/log/k8s-gcp-journald.pos
       read_from_head true
       tag node-journal
+      <storage>
+        @type local
+        path  /var/run/google-fluentd/pos-files/k8s-gcp-journald.pos
```
Comment (igorpeshansky) [inline logging-agent.yaml]: Nit: stray space.

File: logging-agent.yaml [hunk 2]
```diff
@@ -408,43 +408,58 @@ data:
...
 
     <source>
       @type systemd
       filters [{ "_SYSTEMD_UNIT": "node-problem-detector.service" }]
-      pos_file /var/log/k8s-gcp-journald-node-problem-detector.pos
       read_from_head true
       tag node-problem-detector
+      <storage>
+        @type local
+        path /var/run/google-fluentd/pos-files/k8s-gcp-journald-node-problem-detector.pos
```
Comment (igorpeshansky) [inline logging-agent.yaml]: Did this ignore the `pos_file` directive before?

Comment (qingling128) [reply logging-agent.yaml]: Yes, it was ignored before.

**Pull request merged**